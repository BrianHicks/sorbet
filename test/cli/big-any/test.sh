#!/bin/bash
main/sorbet --silence-dev-message --dir test/cli/big-any 2>&1
