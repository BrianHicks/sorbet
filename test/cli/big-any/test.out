big-any.rb:16: Argument does not have asserted type `T::Set[T.class_of(BaseClass)]` https://srb.help/7007
    16 |  x = T.let([A, B, C, D, E, F, G, H, Not].to_set.freeze, T::Set[T.class_of(BaseClass)])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Got `T::Set[T.any(T.class_of(A), T.class_of(B), T.class_of(C), T.class_of(D), T.class_of(E), T.class_of(F), T.class_of(G), T.class_of(H), T.class_of(Not))]` originating from:
    big-any.rb:16:
    16 |  x = T.let([A, B, C, D, E, F, G, H, Not].to_set.freeze, T::Set[T.class_of(BaseClass)])
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Note:
    `T.class_of(Not)` is not a subtype of `T.class_of(BaseClass)`
Errors: 1
